awssso/saml.py

```diff
@@ -43,8 +43,9 @@ class SAMLHelper():
         'duration': ".//a:Assertion/a:AttributeStatement/a:Attribute[@Name='https://aws.amazon.com/SAML/Attributes/SessionDuration']/a:AttributeValue"
     }
 
-    def __init__(self, encoded_payload):
-        self._sts = boto3.client('sts')
+    def __init__(self, aws_profile, encoded_payload):
+        self._session = boto3.Session(profile_name=aws_profile)
+        self._sts = self._session.client('sts')
```

Comment:
Yes, this change lets you use `credential_process` for one profile, but use a separate profile as an anchor point for the STS call. Otherwise boto3 goes down a recursive rabbit hole of `credential_process`. I see you made some other related comments, so we can chat more there!

Reply:
I've run some tests and using something like this works

```
self._sts = boto3.client('sts', aws_access_key_id='', aws_secret_access_key='', aws_session_token='')
```

Reply:
Oh good call, I like that. I was going to suggest creating an explicit `awssso_samlhelper` profile, where the credentials wouldn't even need to be valid. Your way seems cleaner 👍